Explorer/Controller/CreateLogViewController.swift

```diff
@@ -0,0 +1,134 @@
+        
+        UIView.animate(withDuration: 0.34) {
+            self.view.layoutIfNeeded()
+        }
+        
+        switch selectedDatePickerButton {
+            case createLogView.diveDateButton:
+                diveDate = datePickerView.picker.date
+                let formatter = DateFormatter()
+                formatter.dateFormat = "dd MMM yyyy"
+                createLogView.diveDateButton.setTitle("  \(formatter.string(from: diveDate))", for: .normal)
+            case createLogView.startTimeButton:
+                diveStartTime = datePickerView.picker.date
+                let formatter = DateFormatter()
+                formatter.dateFormat = "h:mm a"
+                createLogView.startTimeButton.setTitle("  \(formatter.string(from: diveStartTime))", for: .normal)
+            case createLogView.endTimeButton:
+                diveEndTime = datePickerView.picker.date
+                let formatter = DateFormatter()
+                formatter.dateFormat = "h:mm a"
+                createLogView.endTimeButton.setTitle("  \(formatter.string(from: diveEndTime))", for: .normal)
+        default: ()
+        }
+    }
+    
+    @objc func presentPicker(button: UIButton) {
+        datePickerBottomConstraint?.constant = 0
+        UIView.animate(withDuration: 0.34) {
+            self.view.layoutIfNeeded()
+        }
+        
+        selectedDatePickerButton = button
+        switch selectedDatePickerButton {
+            case createLogView.diveDateButton: datePickerView.configureWith(.date)
+            case createLogView.startTimeButton, createLogView.endTimeButton: datePickerView.configureWith(.time)
+            default: datePickerView.configureWith(.normal)
+        }
+    }
+    
+    @objc func saveDive() {
+        let user = User(fullName: "Jacques Cousteau", username: "JaC", icon: "")
+        let diveNumber = createLogView.diveNumberTextField.text
+        let diveDepth = createLogView.diveDepthSlider.value
+        let diveLocation = Location(name: createLogView.locationTextField.text ?? "Unknown", lat: 33.00, lon: -117.29)
+        var entryType: EntryType = .boat
+        switch createLogView.entryTypeControl.selectedSegmentIndex {
+            case 0: entryType = EntryType.boat
+            default: entryType = EntryType.shore
+        }
+        var waterType: WaterType = .salt
+        switch createLogView.waterTypeControl.selectedSegmentIndex {
+            case 0: waterType = WaterType.salt
+            default: waterType = WaterType.fresh
+        }
+        let diveNotes = createLogView.notesTextView.text
+        let diveBuddies = [User(fullName: "Full Name", username: createLogView.buddiesTextField.text ?? "", icon: "")]
+        
+        let log = DiveLog(diveNumber: Int(diveNumber ?? "0") ?? 0, date: diveDate, depth: Int(diveDepth), location: diveLocation, entryType: entryType, waterType: waterType, timeIn: diveStartTime, timeOut: diveEndTime, notes: diveNotes, user: user, buddies: diveBuddies)
```

Comment:
nit:  It would be much easier to read if you indent this:

```
DiveLog(
    diveNumber: Int(diveNumber ?? "0") ?? 0,
    date: diveDate,
    depth: Int(diveDepth)
    ...
)
```